build: github.com/cybozu-go/netutil@v1.4.0/mtu.go:14:17: undefined: netlink.RouteGet on macOS

Today, we pulled the latest version into my project and got the following error:

```
github.com/cybozu-go/netutil
# github.com/cybozu-go/netutil
../../../../pkg/mod/github.com/cybozu-go/netutil@v1.4.0/mtu.go:14:17: undefined: netlink.RouteGet
../../../../pkg/mod/github.com/cybozu-go/netutil@v1.4.0/mtu.go:16:17: undefined: netlink.RouteGet
```

## go env

```
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="<REDACTED>"
GOENV="<REDACTED>"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="<REDACTED>"
GONOPROXY="<REDACTED>"
GONOSUMDB="<REDACTED>"
GOOS="darwin"
GOPATH="<REDACTED>"
GOPRIVATE="<REDACTED>"
GOPROXY="<REDACTED>"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GOVCS=""
GOVERSION="go1.16"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="<REDACTED>"
CGO_CFLAGS="<REDACTED>"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="<REDACTED>"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="<REDACTED>"
```

## Possible cause

`netlink.RouteGet` is [defined behind `+linux` build tag](https://github.com/vishvananda/netlink/blob/v1.1.0/route_linux.go#L941) but https://github.com/cybozu-go/netutil/blob/v1.4.0/mtu.go#L14 refers to it without the buildtag.

Currently, non-linux environments (e.g. macOS) are unable to compile the module.

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -9,7 +9,10 @@ env:
 jobs:
   test:
     name: Small test
-    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        os: ["ubuntu-20.04", "macos-latest"]
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
@@ -19,3 +22,5 @@ jobs:
       - run: go build .
       - run: go test -race -v -count 1 .
       - run: go vet .
+      - run: cd /tmp; go get honnef.co/go/tools/cmd/staticcheck
+      - run: staticcheck ./...

--- a/halfclose_test.go
+++ b/halfclose_test.go
@@ -2,6 +2,7 @@ package netutil
 
 import (
 	"bytes"
+	"fmt"
 	"io/ioutil"
 	"net"
 	"testing"
@@ -16,12 +17,14 @@ func TestHalfCloser(t *testing.T) {
 		t.Skip(err)
 	}
 
+	errCh := make(chan error, 1)
 	done := make(chan struct{})
 
 	go func() {
+		defer close(done)
 		c, err := l.Accept()
 		if err != nil {
-			t.Error(err)
+			errCh <- err
 			return
 		}
 		defer c.Close()
@@ -33,16 +36,15 @@ func TestHalfCloser(t *testing.T) {
 		c.Write([]byte("test"))
 		err = hc.CloseWrite()
 		if err != nil {
-			t.Fatal(err)
+			errCh <- err
+			return
 		}
 		data, err := ioutil.ReadAll(c)
 		if err != nil {
-			t.Error(err)
+			errCh <- err
+		} else if !bytes.Equal(data, []byte("ack")) {
+			errCh <- fmt.Errorf(`!bytes.Equal(data, []byte("ack")), data=%s`, data)
 		}
-		if !bytes.Equal(data, []byte("ack")) {
-			t.Error(`!bytes.Equal(data, []byte("ack"))`)
-		}
-		close(done)
 	}()
 
 	c, err := net.DialTimeout("tcp", "localhost:15346", 1*time.Second)
@@ -68,4 +70,9 @@ func TestHalfCloser(t *testing.T) {
 	}
 
 	<-done
+	select {
+	case err2 := <-errCh:
+		t.Error(err2)
+	default:
+	}
 }

--- a/mtu.go
+++ b/mtu.go
@@ -1,3 +1,5 @@
+// +build linux
+
 package netutil
 
 import (

--- a/tls.go
+++ b/tls.go
@@ -25,6 +25,7 @@ var (
 	}
 
 	versionMap = map[uint16]string{
+		//lint:ignore SA1019 this is harmless
 		tls.VersionSSL30: "SSL3.0",
 		tls.VersionTLS10: "TLS1.0",
 		tls.VersionTLS11: "TLS1.1",

